# 4.6: Combining & Exporting Data

### This script contains the following points:

#### 1. Import the data sets in Juptyer

#### 2. Check the dimensions of the imported dataframes

#### 3. Merge the dataframes

#### 4. Use the indicator argument to check whether there was a full match between the two dataframes

#### 5. Export data in pickle format

In [1]:
# Importing Libraries

import pandas as pd
import numpy as np
import os

### 1. Import the data sets in Jupyter

In [3]:
#Create File Path
path = r'C:\Users\gre60\Desktop\30-05-2024 Instacart Basket Analysis'

In [2]:
path_2 = r'/Users/gre604/Desktop/30-05-2024 Instacart Basket Analysis'

In [7]:
# Import orders_products_prior data
df_ords_prior = pd.read_csv(os.path.join(path_2, 'Data', 'Original Data', 'orders_products_prior.csv'), index_col = False)

In [6]:
# Import orders_checked data
df_ords_prior = pd.read_csv(os.path.join(path, 'Data', 'Prepared Data', 'orders_checked.csv'), index_col = False)

NameError: name 'path' is not defined

In [8]:
df_ords = pd.read_csv(os.path.join(path_2, 'Data', 'Prepared Data', 'orders_checked.csv'), index_col = False)

### 2. Check the dimensions of the imported dataframes

In [9]:
# Check the output
df_ords_prior.head()

order_id  product_id  add_to_cart_order  reordered
0         2       33120                  1          1
1         2       28985                  2          1
2         2        9327                  3          0
3         2       45918                  4          1
4         2       30035                  5          0

In [10]:
df_ords.head()

Unnamed: 0  order_id  user_id  order_number  orders_day_of_week  \
0           0   2539329        1             1                   2   
1           1   2398795        1             2                   3   
2           2    473747        1             3                   3   
3           3   2254736        1             4                   4   
4           4    431534        1             5                   4   

   order_hour_of_day  days_since_prior_order first_order  
0                  8                     NaN         yes  
1                  7                    15.0          no  
2                 12                    21.0          no  
3                  7                    29.0          no  
4                 15                    28.0          no

In [11]:
df_ords_prior.shape

(32434489, 4)

In [12]:
df_ords.shape

(3421083, 8)

In [13]:
# Drop Unecessary Column
df_ords = df_ords.drop('Unnamed: 0', axis=1)
df_ords

order_id  user_id  order_number  orders_day_of_week  \
0         2539329        1             1                   2   
1         2398795        1             2                   3   
2          473747        1             3                   3   
3         2254736        1             4                   4   
4          431534        1             5                   4   
...           ...      ...           ...                 ...   
3421078   2266710   206209            10                   5   
3421079   1854736   206209            11                   4   
3421080    626363   206209            12                   1   
3421081   2977660   206209            13                   1   
3421082    272231   206209            14                   6   

         order_hour_of_day  days_since_prior_order first_order  
0                        8                     NaN         yes  
1                        7                    15.0          no  
2                       12                    21.0          no  
3                        7                    29.0          no  
4                       15                    28.0          no  
...                    ...                     ...         ...  
3421078                 18                    29.0          no  
3421079                 10                    30.0          no  
3421080                 12                    18.0          no  
3421081                 12                     7.0          no  
3421082                 14                    30.0          no  

[3421083 rows x 7 columns]

### 3. Check the dimensions of the imported dataframes

In [14]:
# Merged orders data

df_merged_large = df_ords.merge(df_ords_prior, on = 'order_id', indicator = True)

In [15]:
df_merged_large.head()

order_id  user_id  order_number  orders_day_of_week  order_hour_of_day  \
0   2539329        1             1                   2                  8   
1   2539329        1             1                   2                  8   
2   2539329        1             1                   2                  8   
3   2539329        1             1                   2                  8   
4   2539329        1             1                   2                  8   

   days_since_prior_order first_order  product_id  add_to_cart_order  \
0                     NaN         yes         196                  1   
1                     NaN         yes       14084                  2   
2                     NaN         yes       12427                  3   
3                     NaN         yes       26088                  4   
4                     NaN         yes       26405                  5   

   reordered _merge  
0          0   both  
1          0   both  
2          0   both  
3          0   both  
4          0   both

In [16]:
df_merged_large.shape

(32434489, 11)

### 4. Use the indicator argument to check whether there was a full match between the two dataframes

In [17]:
df_merged_large['_merge'].value_counts()

both          32434489
right_only           0
left_only            0
Name: _merge, dtype: int64

#### The resulting dataframe (after the merge) has 32,434,489 rows, and each of those rows have information found in both input data sets, as we used an inner join for the purposes of this project.

### 5. Export data in pickle format

In [19]:
# Export data to pkl
df_merged_large.to_pickle(os.path.join(path_2, 'Data','Prepared Data', 'orders_products_combined.pkl'))

In [18]:
# Export modified df_ords dataframe
df_ords.to_csv(os.path.join(path_2, 'Data','Prepared Data', 'orders_final.csv'))